plugins/outputs/groundwork/groundwork.go

```diff
@@ -244,15 +266,18 @@ func (g *Groundwork) parseMetric(metric telegraf.Metric) (string, *transit.Dynam
 				Label:      value.Key + "_cr",
 				Value: &transit.TypedValue{
 					ValueType:   transit.DoubleType,
-					DoubleValue: critical,
+					DoubleValue: &critical,
 				},
 			})
 		}
 
-		typedValue := new(transit.TypedValue)
-		err := typedValue.FromInterface(value.Value)
-		if err != nil {
-			return "", nil, err
+		typedValue := transit.NewTypedValue(value.Value)
+		if typedValue == nil {
+			return "", nil, fmt.Errorf("could not convert value %s: %v", value.Key, value.Value)
+		}
+		if typedValue.ValueType == transit.StringType {
+			g.Log.Warnf("string values are not supported, skipping field %s", value.Key)
+			continue
```

Comment:
Is this still needed with new SDK version?

Also, why just ignore string values and error out on other unsupported values?

Reply:
We assume that TypedValue is created here, so, returning an error if it is not.
And we still need to skip StringType as backend doesn't support it.

Reply:
Okay, but why just logging Warning if type is string and returning error on different unsupported type? I don't get why there's a difference.

For me it should be ignoring in both cases or returning error in both cases..